arelle/ModelValue.py

```diff
@@ -459,18 +571,24 @@ def __str__(self):
             return "PT0S"
         return "P" + ''.join(per)
 
-def time(value, castException=None):
+def time(value: str, castException: Type[Exception] | None = None) -> Time | None:
     if value == "MinTime":
-        return Time(time.min)
+        # Note 2022-09-10
+        # Note sure what this does but don't want to alter code when adding type hints
+        # Ignoring for now
+        return Time(datetime.time.min)
```

Comment:
This requires adding `datetime.time` as an argument of `Time.__new__`. We're left with this error when we call the superclass:
```
Argument 2 to "__new__" of "time" has incompatible type "Union[int, time]"; expected "int"  [arg-type]```

Reply:
Remove `datetime.time` from `Time.__new__`. This code was clearly already busted, so feel free to just have mypy ignore it and I'll write up a followup issue:

If you want to fix it, this should do it:
```python
    if value == "MinTime":
        return Time(
            hour=datetime.time.min.hour,
            minute=datetime.time.min.minute,
            second=datetime.time.min.second,
            microsecond=datetime.time.min.microsecond,
            tzinfo=datetime.time.min.tzinfo,
        )
    elif value == "MaxTime":
        return Time(
            hour=datetime.time.max.hour,
            minute=datetime.time.max.minute,
            second=datetime.time.max.second,
            microsecond=datetime.time.max.microsecond,
            tzinfo=datetime.time.max.tzinfo,
        )
```